### PR TITLE
Test cases for finding a common type with typeMerge()

### PR DESCRIPTION
**Pull request by SSoulaimane**
Every case from [`typeMerge()`](https://github.com/dlang/dmd/blob/b24a87ce4a7e707f6f190ae66a3fe686dc5868a8/src/dmd/dcast.d#L2780) that affects conditional expressions is here, although not every possible combination.

### CODE REVIEW

File: test/compilable/commontype.d [hunk 1]
```diff
@@ -0,0 +1,491 @@
+// REQUIRED_ARGS: -mcpu=avx2
```
Comment (WalterBright) [inline test/compilable/commontype.d]: A comment explaining the purpose of these tests would be nice.

**Review by WalterBright — DISMISSED** [review]
Add explanatory comment.

File: test/compilable/commontype.d (hunk 1 above)
Comment (SSoulaimane) [reply test/compilable/commontype.d]: The purpose is tweaking `typeMerge()` without unintentionally breaking something.

File: test/compilable/commontype.d [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+static assert(Error!( byte16, void16 ));
+static assert(Error!( int4, void16 ));
+static assert(Error!( float4, void16 ));
+static assert(Error!( byte16, ubyte16 ));
+static assert(Error!( short8, ushort8 ));
+static assert(Error!( int4, uint4 ));
+static assert(Error!( int4, float4 ));
+version (D_AVX)
+{
+static assert(Error!( long4, ulong4 ));
```
Comment (JinShil) [inline test/compilable/commontype.d]: What about the types here https://github.com/dlang/druntime/blob/715ccf41b19d64a89e434798acf585005fdd6168/src/core/simd.d#L55-L65?

Comment (SSoulaimane) [reply test/compilable/commontype.d]: I didn't put all of them here, since they're all supposed to fail.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**